tracex_project/tracex/forms.py

```diff
@@ -52,14 +61,23 @@ class BaseEventForm(forms.Form):
         required=True,
         initial=ACTIVITY_KEYS[0][
             0
-        ],  # initialize the activity_key with event_type as key
+        ],
     )
 
     def clean(self):
-        """Validate form data."""
+        """
+        Validate the form by checking that at least one event type or location is selected.
+
+        Raises:
+            forms.ValidationError: If no event type or location is selected.
+
+        Returns:
+            dict: The cleaned data if the validation passes.
+        """
         cleaned_data = super().clean()
-        event_types = cleaned_data.get("event_types")
-        locations = cleaned_data.get("locations")
+        event_types = cleaned_data.get("event_types", [])
```

Comment:
whats the difference in  event_types = cleaned_data.get("event_types") and event_types = cleaned_data.get("event_types", [])?

Reply:
The first one will return `None` if the key `event_types` is not in the `cleaned_data` while the second one provides a default argument for that case and will return an empty list.